File: app/src/main/java/com/test/imageload/MainActivity.java
package com.test.imageload;

import android.os.Bundle;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.test.imageload.imageload.ImageLoad;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.imageView)
    ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        String url = "https://file.yyuehd.com/FuCLS2fyBA4ZcVAHry_meMMG3RW4";
        ImageLoad.loadImage(url, imageView);
    }
}

File: app/src/main/java/com/test/imageload/kotlin/MainActivity.kt
package com.test.imageload.kotlin

import android.content.Intent
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.blankj.utilcode.util.LogUtils
import com.blankj.utilcode.util.ToastUtils
import com.test.imageload.CusZoomViewActivity
import com.test.imageload.R
import com.test.imageload.base.BaseActivity
import com.test.imageload.base.ItemClickListener
import com.test.imageload.db.DbTest
import com.test.imageload.utils.DataUtil
import kotlinx.android.synthetic.main.activity_main.*

/**
 * 使用kotlin 中替换findViewById的方式
 */
class MainActivity : BaseActivity(), ItemClickListener {

    private var adapter: RecyclerViewAdapter? = null

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initData() {
        super.initData()
        var dbTest = DbTest()

        initRecyclerView()
    }

    private fun initRecyclerView() {

        var linearLayoutManager = LinearLayoutManager(act)
        recyclerView.layoutManager = linearLayoutManager
        adapter = RecyclerViewAdapter()
        recyclerView.adapter = adapter
        adapter?.itemClickListener = this

        adapter?.setData(DataUtil.get().generateData())
    }


    /**
     * 条目点击事件
     */
    override fun onItemClick(pos: Int) {
        val posBean = adapter?.getPosBean(pos) ?: return

        val intent = when (pos) {
            0 -> Intent(act, ImageLoadLongPicActivity::class.java)
            1 -> Intent(act, CusZoomViewActivity::class.java)
            2 -> Intent(act, DetailActivity::class.java)
            3 -> Intent(act, VideoCacheActivity::class.java)
            else -> Intent(act, BrowPicActivity::class.java)
        }
        
        act?.startActivity(intent)

        ToastUtils.showShort("我是Item%d,,%s", pos, posBean.name)
    }

    override fun onResume() {
        super.onResume()

    }
}

